cypress-io/cypress
# fix: UNIFY-1408, warnings should be nested appropriately & clearable

- Closes UNIFY-1408, UNIFY-1368 - warnings persisting after dismissal, across testing types
- Closes UNIFY-1604, auto opening global mode project
- Closes UNIFY-1732, clicking into projects does not clear global error

### User facing changelog

Several minor bugfixes, mostly all related to how we're storing error & warning state

### Additional details
- Ensures that warnings are cleared via mutation when the "X" is clicked, rather than maintaining that state on the frontend
- Adds the concept of "layering" the warnings & errors with the part of the lifecycle they're related to, so when switching testing types / projects / browsers, warnings & errors are cleaned up appropriately
- Adds `errorWarningChange` subscription, emitted when error/warning state changes
- Adds concept of "shared" global subscriptions, for state changes that affect both apps

Bonus:
- While adding a test for the warnings & errors, fixed "opening" the project from global mode drop-zone (UNIFY-1604) and clearing the error when going back to the global projects list (UNIFY-1732)
- Fixes for a few other flaky tests

### Steps to test
- Trigger a warning, by having `experimentalStudio` set as a config prop, or a `baseUrl` on a port that is not running,  observe the warning can be cleared

### How has the user experience changed?


### PR Tasks
